internal/checks/checks.go

```
@@ -0,0 +1,177 @@
+	"fmt"
+	"slices"
+
+	"github.com/NETWAYS/go-check"
+	"github.com/NETWAYS/go-check/perfdata"
+	"github.com/NETWAYS/go-check/result"
+	"github.com/icinga/check-netgear/internal/utils"
+	"github.com/icinga/check-netgear/netgear"
+)
+
+// CheckCPU creates a partialResult with the CPU information
+func CheckCPU(cpuUsage float64, warn float64, crit float64) (*result.PartialResult, error) {
+	status := utils.StatusByThreshold(cpuUsage, warn, crit)
+	partial := result.PartialResult{
+		Output: fmt.Sprintf("CPU Usage: %.2f%%", cpuUsage),
+	}
+	if err := partial.SetState(status); err != nil {
+		return nil, err
+	}
+	partial.Perfdata.Add(&perfdata.Perfdata{Label: "CPU", Value: cpuUsage, Min: 0, Max: 100})
+	return &partial, nil
+}
+
+// CheckMemory creates a partialResult with the memory information
+func CheckMemory(memUsage float64, warn float64, crit float64) (*result.PartialResult, error) {
+	status := utils.StatusByThreshold(memUsage, warn, crit)
+	partial := result.PartialResult{
+		Output: fmt.Sprintf("RAM Usage: %.2f%%", memUsage),
+	}
+	if err := partial.SetState(status); err != nil {
+		return nil, err
+	}
+	partial.Perfdata.Add(&perfdata.Perfdata{Label: "RAM", Value: memUsage, Min: 0, Max: 100})
+	return &partial, nil
+}
+
+// CheckTemperature creates a partialResult with the temperature information
+func CheckTemperature(sensors []netgear.SensorDetail, warn float64, crit float64) (*result.PartialResult, error) {
+	partial := result.PartialResult{Output: "Temperature"}
+	worst := check.OK
+
+	for _, s := range sensors {
+		status := utils.StatusByThreshold(s.Temperature, warn, crit)
+		if status > worst {
+			worst = status
+		}
+
+		sub := result.PartialResult{
+			Output: fmt.Sprintf("%s: %.1f°C", s.Description, s.Temperature),
+		}
+		if err := sub.SetState(status); err != nil {
+			return nil, err
+		}
+		sub.Perfdata.Add(&perfdata.Perfdata{Label: s.Description, Value: s.Temperature, Min: 0})
+		partial.AddSubcheck(sub)
+	}
+
+	_ = partial.SetState(worst)
```

You're still ignoring errors here :)